File: Wizard/ClientApp/src/app/wizard.service.ts
import { Step } from './models/Step';
import { Injectable, Inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Item } from './models/Item';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class WizardService {

  private apiEndPoint = this.baseUrl + 'Wizard/';

  constructor(private http: HttpClient, @Inject('BASE_URL') private baseUrl: string) {}

  public GetSteps():Observable<Step[]>{
    return this.http.get<Step[]>(this.apiEndPoint + 'GetSteps');
  }

  public GetItems(stepId: string):Observable<Item[]>{
    return this.http.get<Item[]>(this.apiEndPoint + 'GetItems?stepId='+ stepId);
  }

  public AddStep():Observable<Step>{
    return this.http.post<Step>(this.apiEndPoint + 'AddStep', null);
  }

  public RemoveStep(id: string):Observable<any>{
    return this.http.delete(this.apiEndPoint + 'RemoveStep?id='+ id);
  }

  public AddItem(item: Item):Observable<Item>{
    return this.http.post<Item>(this.apiEndPoint + 'AddItem', item);
  }

  public EditItem(item: Item):Observable<Item>{
    return this.http.post<Item>(this.apiEndPoint + 'EditItem', item);
  }

  public RemoveItem(id: string, stepId: string):Observable<any>{
    return this.http.delete(`${this.apiEndPoint}RemoveItem?id=${id}&stepId=${stepId}`);
  }
}

File: Wizard/Services/WizardService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Wizard.Data.EFCore;
using Wizard.Models;

namespace Wizard.Services
{
    public class WizardService : IWizardService
    {
        private readonly EfCoreItemRepository efCoreItemRepository;
        private readonly EfCoreStepRepository efCoreStepRepository;

        public WizardService(EfCoreItemRepository efCoreItemRepository, EfCoreStepRepository efCoreStepRepository)
        {
            this.efCoreItemRepository = efCoreItemRepository;
            this.efCoreStepRepository = efCoreStepRepository;
        }

        public async Task<List<Step>> GetSteps()
        {
            return await efCoreStepRepository.GetAll();
        }

        public async Task<Step> AddStep()
        {
            var step = new Step { Id = Guid.NewGuid() };

            return await efCoreStepRepository.Add(step);
        }

        public async Task RemoveStep(Guid id)
        {
            await efCoreStepRepository.Delete(id);
        }

        public async Task<List<Item>> GetItems(Guid stepId)
        {
            return await efCoreItemRepository.GetItemsByStepId(stepId);
        }

        public async Task<Item> AddItem(Item item)
        {
            item.Id = Guid.NewGuid();

            return await efCoreItemRepository.Add(item);
        }

        public async Task<Item> EditItem(Item item)
        {
            return await efCoreItemRepository.Update(item);
        }

        public async Task RemoveItem(Guid id, Guid stepId)
        {
            await efCoreItemRepository.Delete(id, stepId);
        }
    }
}
